### PR TITLE
Add support for number of interval pieces to diffForHumans

### PR DESCRIPTION
**Pull request by pushpak**
Example Usage:
`Carbon::now()->addMinute()->addSecond()->diffForHumans(null, true,
false, 2);`
returns **1 minute 1 second**

`Carbon::now()->addYears(2)->addMonths(3)->addDay()->addSecond()->diffForHumans(null, true, true, 4);`
returns **2yrs 3mos 1d 1s**

### CODE REVIEW

**Comment by lucasmichot** [comment]
Your tests fail @pushpak : https://travis-ci.org/briannesbitt/Carbon/builds/159898184

**Comment by pushpak** [comment]
@lucasmichot should be good now https://travis-ci.org/briannesbitt/Carbon/builds/159943436 👍

**Comment by pushpak** [comment]
@lucasmichot all done 👍

**Comment by jbrooksuk** [comment]
Our use case for this feature is that we have objects which expire in say `1 week and 6 days` but Carbon is displaying `1 week` which is incorrect.

**Comment by jbrooksuk** [comment]
Actually, this doesn't work. If you have a date which is 1 week and 6 days, the output is like `1 week and 10 hours`.

**Comment by pushpak** [comment]
@jbrooksuk added more tests. Works fine here. And rebased also.

**Comment by lucasmichot** [comment]
@pushpak can you rebase please?

**Comment by pushpak** [comment]
@lucasmichot Done

**Comment by lucasmichot** [comment]
@jbrooksuk is your edge case covered?

**Comment by jbrooksuk** [comment]
I'll take a look now, bear with me.

**Comment by jbrooksuk** [comment]
`Carbon::parse('2017-01-26')->diffForHumans(null, true, false, 2)` is returning 1 week 8 hours.

Shouldn't this be 1 week and 6 days still?

**Comment by lucasmichot** [comment]
Indeed

**Comment by jbrooksuk** [comment]
It's a 👎 then sorry.

**Comment by pushpak** [comment]
@jbrooksuk sorry missed it. Should be working now. Please check.

@lucasmichot its fixed now.

**Comment by lucasmichot** [comment]
Can you squash your commits @pushpak ?

**Comment by pushpak** [comment]
@lucasmichot all Done

File: src/Carbon/Carbon.php [hunk 1]
```diff
@@ -2783,61 +2783,72 @@ public function secondsUntilEndOfDay()
...
-            case $diffInterval->d > 0:
-                $unit = $short ? 'd' : 'day';
-                $count = $diffInterval->d;
+        foreach ($diffIntervalArray as $diffIntervalData) {
+            if ($diffIntervalData['value'] > 0) {
+                $unit = $short ? $diffIntervalData['unitShort'] : $diffIntervalData['unit'];
+                $count = $diffIntervalData['value'];
 
-                if ($count >= static::DAYS_PER_WEEK) {
+                if ($diffIntervalData['unit'] == 'day' && $count >= static::DAYS_PER_WEEK) {
```
Comment (lucasmichot) [inline src/Carbon/Carbon.php]: Any reason not to use a stricter `===` comparison?

File: src/Carbon/Carbon.php [hunk 2]
```diff
@@ -2847,10 +2858,13 @@ public function diffForHumans(Carbon $other = null, $absolute = false, $short =
 
         $transId = $isNow ? ($isFuture ? 'from_now' : 'ago') : ($isFuture ? 'after' : 'before');
 
-        // Some langs have special pluralization for past and future tense.
-        $tryKeyExists = $unit.'_'.$transId;
-        if ($tryKeyExists !== static::translator()->transChoice($tryKeyExists, $count)) {
-            $time = static::translator()->transChoice($tryKeyExists, $count, array(':count' => $count));
+        if ($parts === 1) {
+            // Some langs have special pluralization for past and future tense.
+            $key = $unit.'_'.$transId;
```
Comment (lucasmichot) [inline src/Carbon/Carbon.php]: `$unit` might not be set at this point, so it probably needs to be initiated somewhere earlier in the function?

File: src/Carbon/Carbon.php (hunk 1 above)
Comment (pushpak) [inline src/Carbon/Carbon.php]: done

File: src/Carbon/Carbon.php (hunk 2 above)
Comment (pushpak) [inline src/Carbon/Carbon.php]: done

**Pull request merged**